### PR TITLE
Adding Cypress UI Tests

### PR DESCRIPTION
**Pull request by cshnimble**

### CODE REVIEW

**Review by dangood84 — DISMISSED** [review]

File: Dfe.Academies.External.Web/CypressTests/cypress.config.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+/*const {
+  username,
+  password
+} = require('./config')
+*/
+
+
+module.exports = defineConfig({
+  e2e: {
+    url: 'https://s184t01-a2bextcdnendpoint-ezfhhdbpembpanh5.z01.azurefd.net',
```
Comment (cshnimble) [inline Dfe.Academies.External.Web/CypressTests/cypress.config.js:11]: ⚠️ As mentioned above, these all need to be removed from the code before merge and become environment variables

File: Dfe.Academies.External.Web/CypressTests/config.ts [hunk 2]
```diff
@@ -0,0 +1,5 @@
+export const url = Cypress.env('url')
+export const login_username = Cypress.env('username')
+export const login_password = Cypress.env('password')
+
+export const unAuthUrl = "https://s184t01-a2bextcdnendpoint-ezfhhdbpembpanh5.z01.azurefd.net/application-overview?appId=10021"
```
Comment (cshnimble) [inline Dfe.Academies.External.Web/CypressTests/config.ts:5]: Make this an env var as above

File: Dfe.Academies.External.Web/CypressTests/package.json
Comment (cshnimble) [inline Dfe.Academies.External.Web/CypressTests/package.json:1]: Can you update this with the proper information, scripts etc? See [prepare conversions](https://github.com/DFE-Digital/prepare-academy-conversions/blob/main/Dfe.PrepareConversions/Dfe.PrepareConversions.CypressTests/package.json) as an example

File: Dfe.Academies.External.Web/CypressTests/package.json [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  "description": "",
+  "main": "index.js",
+  "scripts": {
+    "cy:open": "cypress open",
+    "cy:run": "cypress run"
+  },
+  "keywords": [],
+  "author": "",
+  "license": "ISC",
+  "dependencies": {
```
Comment (cshnimble) [inline Dfe.Academies.External.Web/CypressTests/package.json:13]: I think you're missing typescript from the dependencies?

File: Dfe.Academies.External.Web/CypressTests/cypress/page-objects/pages/A2BLogin.ts
Comment (cshnimble) [inline Dfe.Academies.External.Web/CypressTests/cypress/page-objects/pages/A2BLogin.ts:1]: @dangood84 do we need to be checking the different login error cases? I think this is all owned by dfe sign-in who should have checked all these themselves so we can probably just use login normally in our tests

**Comment by cshnimble** [comment]
Can you add information to the README to explain how to run the tests?

Will also be worth adding eslint to the tests at some point (can be done at a later time)

File: Dfe.Academies.External.Web/CypressTests/cypress/fixtures/data-generator.ts
Comment (cshnimble) [inline Dfe.Academies.External.Web/CypressTests/cypress/fixtures/data-generator.ts:1]: I'd bin this off and use a package like [Faker](https://fakerjs.dev/) if you need to generate data

File: Dfe.Academies.External.Web/CypressTests/cypress/fixtures/example.json
Comment (cshnimble) [inline Dfe.Academies.External.Web/CypressTests/cypress/fixtures/example.json:1]: Can delete this file

File: Dfe.Academies.External.Web/CypressTests/cypress/e2e/createNewJAMApp.cy.ts [hunk 4]
```diff
@@ -0,0 +1,1079 @@
...
+       Footer.accessibilityStatementLinkVisible()
+       Footer.cookiesLinkVisible()
+       Footer.termsAndConditionsLinkVisible()
+       Footer.privacyLinkVisible()
+       Footer.oglLogoVisible()
+       Footer.allContentTextVisible()
+       Footer.openGovernmentLicence3LinkVisible()
+       Footer.crownCopyrightLinkVisible()
+
+       cy.wait(2000)
```
Comment (cshnimble) [inline Dfe.Academies.External.Web/CypressTests/cypress/e2e/createNewJAMApp.cy.ts:177]: I'd avoid explicit waits as much as possible, instead opting for an implicit wait

See docs [here](https://docs.cypress.io/api/commands/wait#Time)

**Pull request merged**